spec/elastic_apm/metadata/system_info_spec.rb

```diff
@@ -34,6 +34,16 @@ module ElasticAPM
         it 'has no newline at the end' do
           expect(subject.detected_hostname).not_to match(/\n\z/)
         end
+
+        context 'when there is an exception' do
+          before do
+            allow(Socket).to receive(:gethostname).and_raise(StandardError)
+          end
+
+          it 'returns an empty string' do
+            expect(subject.detected_hostname).to eq('')
```

Comment:
Why empty string and not `nil`? The latter would skip the whole field in the serializer code above.

Reply:
I came to the same conclusion but forgot to update the test. The `detected_hostname` method returns `nil` when there's an exception.